### ISSUE TITLE
Schema generation sets null in definition if model used more than once

### ISSUE THREAD

**Issue by josepaiva94** [samuelcolvin/pydantic#601]
# Bug
When a custom model is referred more than once in the root model, its definition becomes **null** in the generated JSON Schema.

* OS: **?** Ubuntu  18
* Python version `import sys; print(sys.version)`: **?** 3.7.1
* Pydantic version `import pydantic; print(pydantic.VERSION)`: **?** 0.28

```py
from typing import List
from pydantic import BaseModel


class Foo(BaseModel):
    count: int = 0
    size: float = None


class Bar(BaseModel):
    apple = 'x'
    banana = 'y'


class FooBar(BaseModel):
    foo: Foo
    bar: Bar


class FooFoosBar(BaseModel):
    foo: Foo
    foos: List[Foo]
    bar: Bar


print(FooBar.schema_json(indent=2))
print('------------------------------')
print(FooFoosBar.schema_json(indent=2))
```

Output
```
{
  "title": "FooBar",
  "type": "object",
  "properties": {
    "foo": {
      "$ref": "#/definitions/Foo"
    },
    "bar": {
      "$ref": "#/definitions/Bar"
    }
  },
  "required": [
    "foo",
    "bar"
  ],
  "definitions": {
    "Foo": {
      "title": "Foo",
      "type": "object",
      "properties": {
        "count": {
          "title": "Count",
          "default": 0,
          "type": "integer"
        },
        "size": {
          "title": "Size",
          "type": "number"
        }
      }
    },
    "Bar": {
      "title": "Bar",
      "type": "object",
      "properties": {
        "apple": {
          "title": "Apple",
          "default": "x",
          "type": "string"
        },
        "banana": {
          "title": "Banana",
          "default": "y",
          "type": "string"
        }
      }
    }
  }
}
------------------------------
{
  "title": "FooFoosBar",
  "type": "object",
  "properties": {
    "foo": {
      "$ref": "#/definitions/Foo"
    },
    "foos": {
      "title": "Foos",
      "type": "array",
      "items": {
        "$ref": "#/definitions/Foo"
      }
    },
    "bar": {
      "$ref": "#/definitions/Bar"
    }
  },
  "required": [
    "foo",
    "foos",
    "bar"
  ],
  "definitions": {
    "Foo": null,
    "Bar": {
      "title": "Bar",
      "type": "object",
      "properties": {
        "apple": {
          "title": "Apple",
          "default": "x",
          "type": "string"
        },
        "banana": {
          "title": "Banana",
          "default": "y",
          "type": "string"
        }
      }
    }
  }
}
```

Expected Output
```
{
  "title": "FooBar",
  "type": "object",
  "properties": {
    "foo": {
      "$ref": "#/definitions/Foo"
    },
    "bar": {
      "$ref": "#/definitions/Bar"
    }
  },
  "required": [
    "foo",
    "bar"
  ],
  "definitions": {
    "Foo": {
      "title": "Foo",
      "type": "object",
      "properties": {
        "count": {
          "title": "Count",
          "default": 0,
          "type": "integer"
        },
        "size": {
          "title": "Size",
          "type": "number"
        }
      }
    },
    "Bar": {
      "title": "Bar",
      "type": "object",
      "properties": {
        "apple": {
          "title": "Apple",
          "default": "x",
          "type": "string"
        },
        "banana": {
          "title": "Banana",
          "default": "y",
          "type": "string"
        }
      }
    }
  }
}
------------------------------
{
  "title": "FooFoosBar",
  "type": "object",
  "properties": {
    "foo": {
      "$ref": "#/definitions/Foo"
    },
    "foos": {
      "title": "Foos",
      "type": "array",
      "items": {
        "$ref": "#/definitions/Foo"
      }
    },
    "bar": {
      "$ref": "#/definitions/Bar"
    }
  },
  "required": [
    "foo",
    "foos",
    "bar"
  ],
  "definitions": {
    "Foo": {
      "title": "Foo",
      "type": "object",
      "properties": {
        "count": {
          "title": "Count",
          "default": 0,
          "type": "integer"
        },
        "size": {
          "title": "Size",
          "type": "number"
        }
      }
    },
    "Bar": {
      "title": "Bar",
      "type": "object",
      "properties": {
        "apple": {
          "title": "Apple",
          "default": "x",
          "type": "string"
        },
        "banana": {
          "title": "Banana",
          "default": "y",
          "type": "string"
        }
      }
    }
  }
}
```

**Comment by samuelcolvin** [comment]
@tiangolo any ideas?

**Comment by tiangolo** [comment]
Hmm, not sure. I'll check it as soon as I can.

**Comment by tiangolo** [comment]
This was fixed in #621. :tada: 

It was fixed and is available since a very old release (although not sure which). But it certainly works correctly in the lastest (v1.5.1). :rocket:
